// event.proto

syntax = "proto3";

package event;

message Event {
  string event_id = 1;
  string event_type = 2;
  string sender_id = 3;
  string receiver_id = 4;
  string text = 5;
  string metadata = 6;
  string timestamp = 7;
}

message EventBatch {
  repeated Event events = 1;
}

message Ack {
  bool success = 1;
  string message = 2;
}

service EventService {
  rpc SendEvent(Event) returns (Ack);
  rpc SendEvents(EventBatch) returns (Ack);
}